feat: angular + sanity template updates

osnoser1 opened the pull request:
## Summary
This PR updates the Angular + Sanity template with modern features, including Angular 19 migration, improved workflow configurations, and optimized component architecture.

## Changes

### Angular Updates
- Migrated to Angular 19.2.2
- Changed from module-based to standalone components
- Updated component architecture to use OnPush change detection
- Removed app-routing module in favor of modern routing approach
- Implemented resource-based data fetching
- Added proper typing for Sanity content
- Zoneless application

### Asset Management
- Moved assets from `src/assets` to `public/assets`
- Relocated SVG files to follow modern Angular conventions

### Sanity Integration
- Improved environment variable handling
- Updated Sanity dependencies to latest versions
- Added clearer instructions for project setup

### Build and Development
- Updated build configuration in angular.json
- Modified dev scripts to use npm-run-all for better parallel execution
- Updated Node engine requirements to >= 20.11
- Updated ESLint configuration for Sanity Studio

### Documentation
- Added clearer instructions for copying project ID and dataset
- Improved deployment instructions
- Updated README with current best practices

### Build results

<img width="580" alt="image" src="https://github.com/user-attachments/assets/e7e1bea5-bf79-42c7-8f5d-e6b2689b24fe" />

### Additional notes

- The output folder changed, now. the bundles are in `dist/browser`, I suppose you'll have to update it in vercel 
<img width="271" alt="image" src="https://github.com/user-attachments/assets/67a3fb09-668a-467a-a4d5-3535a5725d15" />

## Related Issues
- Closes #13

osnoser1 committed:
build: remove unused ng-universal

osnoser1 committed:
build: update to a17

osnoser1 committed:
build: update to a18

osnoser1 committed:
build: update to a19

osnoser1 committed:
refactor: standalone components

osnoser1 committed:
refactor: signal inputs

osnoser1 committed:
fix: base href

osnoser1 committed:
refactor: align public folder config

osnoser1 committed:
refactor: built-in control flow syntax

osnoser1 committed:
chore: remove unused css file

osnoser1 committed:
build: update dependencies

osnoser1 committed:
chore: improve readme instructions

osnoser1 committed:
refactor: lazy loading in components

osnoser1 committed:
refactor: reduce sanity service code

schmetti-dev commented:
thanks for that @osnoser1! helped me get mine working with angular v19 too

osnoser1 commented:
Just a friendly reminder to check this PR when you can @kenjonespizza; I'd like to make improvements to this template but it should be ideal to merge this PR first 😄 

Thank you very much in advance

kenjonespizza commented on angular-app/src/environments/setEnvVars.ts:
```diff
@@ -1,26 +1,23 @@
-  require("dotenv").config({
-    path: "./.env",
-  });
+  process.loadEnvFile('./.env');
```
👋 @osnoser1 Im getting a Vercel build error with this.

``` 
Error: ENOENT: no such file or directory, open './.env'
    at process.loadEnvFile (node:internal/process/per_thread:320:7)
    at setEnv (/vercel/path0/angular-app/src/environments/setEnvVars.ts:8:11)
    at Object.<anonymous> (/vercel/path0/angular-app/src/environments/setEnvVars.ts:49:1)
    at Module._compile (node:internal/modules/cjs/loader:1730:14)
    at Object..js (node:internal/modules/cjs/loader:1895:10)
    at Module.load (node:internal/modules/cjs/loader:1465:32)
    at Function._load (node:internal/modules/cjs/loader:1282:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: './.env'
}
```

```
npm error Lifecycle script `setEnvVars` failed with error:
npm error code 1
npm error path /vercel/path0/angular-app
npm error workspace angular-app
npm error location /vercel/path0/angular-app
npm error command failed
npm error command sh -c node src/environments/setEnvVars.ts
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /vercel/path0/angular-app
npm error workspace angular-app
npm error location /vercel/path0/angular-app
npm error command failed
npm error command sh -c npm run setEnvVars && ng build
Error: Command "npm run build" exited with 1
```

kenjonespizza reviewed (commented):
Hi @osnoser1 👋 !  I don't think an apology even starts to cover my feelings for this taking so long.

With the Spring launch and post launch materials wrapping up, I've prioritized this.  From what I can tell this PR looks Amazing and works for me 😊.  The only thing Im stuck on is the Vercel deployment.  I'm not sure why this isn't working.  I do have the env vars set in Vercel, but I suspect this has something to do with how Angular generates the env file?  Open to your thoughts

osnoser1 committed:
fix: handles missing .env file gracefully

Modifies the application to handle the case where a `.env` file is not present, such as in production environments.

Instead of throwing an error, it logs a message and proceeds to use environment variables set by the deployment platform. This allows the application to run without a `.env` file.

osnoser1 commented:
> Hi @osnoser1 👋 ! I don't think an apology even starts to cover my feelings for this taking so long.
> 
> With the Spring launch and post launch materials wrapping up, I've prioritized this. From what I can tell this PR looks Amazing and works for me 😊. The only thing Im stuck on is the Vercel deployment. I'm not sure why this isn't working. I do have the env vars set in Vercel, but I suspect this has something to do with how Angular generates the env file? Open to your thoughts

Hi @kenjonespizza, I completely understand, no worries, I'd like to send another PR updating to v20 in a couple of days 😄 .
I think I fixed, please, could you run again the CI in vercel? Please 😄 

I changed dotenv to the native version of node to load env files, however the the method throws an exception if the files isn't found, and dotenv just ignore that.

I'm pending, let me know anything 😄

kenjonespizza reviewed (approved):
Looking good!  Thank you!!
